codeforamerica/vita-min
# Adds persona import/export as a CLI

## Link to pivotal/JIRA issue
- https://codeforamerica.atlassian.net/browse/FYST-1216

## Is PM acceptance required? (delete one)
- No - merge after code review approval

**Reminder**: merge main into this branch and get green tests before merging to main

## What was done?
- This adds both a `personas:export` and a `personas:import` command. No tests included, due to being timeboxed, but if it looks good I'd like to polish it.

## How to test?
The two commands can be run locally to test. Just find an intake with a json blob and xml blob that you want to export as a persona. For example, for a `StateFileIdIntake` you can export either the xml or the json by passing either the `x` or `j` flag as appropriate.

`bin/thor personas:export --xml --state id 30` or `bin/thor personas:export -xs id 30` for a little more brevity.

This will output to stdout which can conveniently be imported with `bin/thor personas:import -s id <some_persona_slug>`. Note that the file extension is not given, it will detect if it is a json blob or an xml blob automatically by checking if it begins with an angle bracket.

The idea is that you can do something like:

`aptible ssh --app vita-app-demo bin/thor personas:export -xs id 30 | bin/thor personas:import -s id penelope_pipsqueak` to get the xml added.

and

`aptible ssh --app vita-app-demo bin/thor personas:export -js id 30 | bin/thor personas:import -s id penelope_pipsqueak`  to get the json persona added.

Note the only difference between the commands is whether an xml or json flag is passed. I was racking my brain trying to think of a way to get both at once but I think it being split between local and remote unfortunately necessitates two runs.

## Screenshots (for visual changes)
- Before
- After
